che-incubator/che-code
# Alignment with upstream

### What does this PR do?
- Rebase against the upstream
- we still can not align Node.js version with upstream as 20.18.1 version is not available in the images that we use to prepare che-code assembly

### What issues does this PR fix?

https://github.com/eclipse-che/che/issues/23252

### How to test this PR?
- PR jobs should be green
- testing of any use cases is useful as we pick up a lot of commits from the upstream

You can use the following links to start a workspace for testing:
- `ubi8-based` tools container: https://che-dogfooding.apps.che-dev.x6e0.p1.openshiftapps.com/#/https://github.com/RomanNikitenko/web-nodejs-sample/tree/ubi8-based
- `ubi8-nodejs-based` tools container: https://che-dogfooding.apps.che-dev.x6e0.p1.openshiftapps.com/#/https://github.com/RomanNikitenko/web-nodejs-sample/tree/ubi8-nodejs-based
- `ubi9-based` tools container: https://che-dogfooding.apps.che-dev.x6e0.p1.openshiftapps.com/#/https://github.com/RomanNikitenko/web-nodejs-sample/tree/ubi9-based
- `ubi9-nodejs-based` tools container: https://che-dogfooding.apps.che-dev.x6e0.p1.openshiftapps.com/#/https://github.com/RomanNikitenko/web-nodejs-sample/tree/ubi9-nodejs-based

### Does this PR contain changes that override default upstream Code-OSS behavior?
